repo/packages/A/alluxio-enterprise/100/resource.json

```diff
@@ -0,0 +1,62 @@
+{
+  "assets": {
+    "uris": {
+      "alluxio-agent-darwin_386": "https://alluxio-dcos-release.s3.amazonaws.com/agent/v1.5.0/darwin_386/alluxio-dcos",
+      "alluxio-agent-darwin_amd64": "https://alluxio-dcos-release.s3.amazonaws.com/agent/v1.5.0/darwin_amd64/alluxio-dcos",
+      "alluxio-agent-linux_386": "https://alluxio-dcos-release.s3.amazonaws.com/agent/v1.5.0/linux_386/alluxio-dcos",
+      "alluxio-agent-linux_amd64": "https://alluxio-dcos-release.s3.amazonaws.com/agent/v1.5.0/linux_amd64/alluxio-dcos",
+      "bootstrap-zip": "https://alluxio-dcos-release.s3.amazonaws.com/artifacts/2.1.1-1.7.1/bootstrap.zip",
+      "custom-download-url": "",
```

Comment:
FYI The failing CI is due to this empty asset uri

Reply:
@takirala Thanks for your response. I need some feedback. 

Here is the flow of how the alluxio-enterprise tarball is downloaded:
1. check `assets.custom-download-url`.
1.a. If not empty, download url using `curl`. This is the path used for air-gapped environments. 
1.b If empty, download the platform specific go agent binary from `assets.alluxio-agent-*`

The agent binary uses `service.license` specified as part of the installation config json. The agent uses the license to authenticate against an Alluxio proxy server to get the alluxio-enterprise tarball as it is not hosted publicly.

What do you suggest?. Either you fix the build scripts to accept empty assets or let me know to handle such an authenticated download

Reply:
@madanadit 
I am not sure who/what would set this url to be non-empty. Does someone edit the `resource.json` while building packages for local universe ? If yes, what do you think of moving this `custom_download_url` to `config.json` (for e.g.: `service.alluxio.uris.custom-download-url`) and then you can use that in the marathon mustache. Like : 

Change the marathon mustache from : 
```
TASKCFG_ALL_CUSTOM_DOWNLOAD_URL": "{{resource.assets.uris.custom-download-url}}"
```
to 
```
TASKCFG_ALL_CUSTOM_DOWNLOAD_URL": "{{service.alluxio.uris.custom-download-url}}"
```
In this the case the someone (or something) can look at the config.json and change the value instead of looking at `resource.json`. The `service.license` part would still be unaffected. Let me know if this works for you or if I overlooked something.

cc : @ryadav88

Reply:
correct, currently `assets.custom-download-url` is set while building the local universe. previously the recommendation i got was to specify all required urls as part of `resources.assets`. the rationale was to have a single location for specifying all assets that need to be downloaded

Reply:
Even with the current structure of `resource.json` and mustache template, if "someone" doesn't specify the `assets.custom-download-url` during the building of local universe, the deployment would still start and then fail. So you are not loosing anything here right? 

    ...instead of an informative message up front about the required parameter. 

Who will throw this error? I am _assuming_ the installer would look at `TASKCFG_ALL_CUSTOM_DOWNLOAD_URL` and throw based on its presence? If yes, from the perspective of installer, all it cares is the value of `TASKCFG_ALL_CUSTOM_DOWNLOAD_URL` and as long as it is set from `config.json` OR `resource.json`, nothing else should change. Just to make myself clear, what I suggested was instead of someone editing the `resource.json`, they would edit the `config.json` now before building the local universe and the rest of the flow would still be the same.

As for your other concern on all urls being at one place, i would argue that, although the urls look more dispersed, this structure makes more sense as `assets` are well, meant to be dumb downloadable URLs. `custom_download_url : ""` brings some _sort_ of business logic in to that. we want to avoid this by moving it `config.json` or `marathon template` where the logic should reside. 

Although a long shot, I would add that by moving the value to config.json, you are giving the user a more easy way to override `custom_download_url` from the DC/OS UI if needed!

Reply:
thanks for the clarification!. i mis-understood the part about modifying `config.json` before building the local universe. i'll make the change.